From 6dd25eb3ca3d8a6847fab029946d15f223b1a551 Mon Sep 17 00:00:00 2001
From: Nikita Popov <npopov@redhat.com>
Date: Mon, 31 Jan 2022 10:33:27 +0100
Subject: [PATCH] Disable CrashRecoveryTest.DumpStackCleanup test on aarch64

Produces a non-reproducible failure on aarch64:
https://bugzilla.redhat.com/show_bug.cgi?id=2048440
---
 llvm/unittests/Support/CrashRecoveryTest.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/llvm/unittests/Support/CrashRecoveryTest.cpp b/llvm/unittests/Support/CrashRecoveryTest.cpp
index d942c9d..28efedd 100644
--- a/llvm/unittests/Support/CrashRecoveryTest.cpp
+++ b/llvm/unittests/Support/CrashRecoveryTest.cpp
@@ -76,6 +76,7 @@ TEST(CrashRecoveryTest, Cleanup) {
   llvm::CrashRecoveryContext::Disable();
 }
 
+#ifndef __aarch64__
 TEST(CrashRecoveryTest, DumpStackCleanup) {
   SmallString<128> Filename;
   std::error_code EC = sys::fs::createTemporaryFile("crash", "test", Filename);
@@ -101,6 +102,7 @@ TEST(CrashRecoveryTest, DumpStackCleanup) {
   EXPECT_EQ(GlobalInt, 1);
   llvm::CrashRecoveryContext::Disable();
 }
+#endif
 
 TEST(CrashRecoveryTest, LimitedStackTrace) {
   // FIXME: Handle "Depth" parameter in PrintStackTrace() function
-- 
1.8.3.1

